test/functional/test_runner.py

```diff
@@ -479,33 +479,35 @@ def run_tests(*, test_list, src_dir, build_dir, tmpdir, jobs=1, attempts=1, enab
 
     max_len_name = len(max(test_list, key=len))
     test_count = len(test_list)
-    for i in range(test_count):
-        test_result, testdir, stdout, stderr = job_queue.get_next()
-        test_results.append(test_result)
-        done_str = "{}/{} - {}{}{}".format(i + 1, test_count, BOLD[1], test_result.name, BOLD[0])
-        if test_result.status == "Passed":
-            logging.debug("%s passed, Duration: %s s" % (done_str, test_result.time))
-        elif test_result.status == "Skipped":
-            logging.debug("%s skipped" % (done_str))
-        else:
-            print("%s failed, Duration: %s s\n" % (done_str, test_result.time))
-            print(BOLD[1] + 'stdout:\n' + BOLD[0] + stdout + '\n')
-            print(BOLD[1] + 'stderr:\n' + BOLD[0] + stderr + '\n')
-            if combined_logs_len and os.path.isdir(testdir):
-                # Print the final `combinedlogslen` lines of the combined logs
-                print('{}Combine the logs and print the last {} lines ...{}'.format(BOLD[1], combined_logs_len, BOLD[0]))
-                print('\n============')
-                print('{}Combined log for {}:{}'.format(BOLD[1], testdir, BOLD[0]))
-                print('============\n')
-                combined_logs_args = [sys.executable, os.path.join(tests_dir, 'combine_logs.py'), testdir]
-                if BOLD[0]:
-                    combined_logs_args += ['--color']
-                combined_logs, _ = subprocess.Popen(combined_logs_args, universal_newlines=True, stdout=subprocess.PIPE).communicate()
-                print("\n".join(deque(combined_logs.splitlines(), combined_logs_len)))
-
-            if failfast:
-                logging.debug("Early exiting after test failure")
-                break
+    i = 0
+    while i < test_count:
```

Comment:
Probably you need also `    Merge bitcoin/bitcoin#24195: test: Fix failfast option for functional test runner` otherwise --failfast will be broken

Reply:
I broke intentionally one test, after that your branch:

```
Early exiting after test failure
3/179 - wallet_backup.py passed, Duration: 30 s
4/179 - feature_dip3_deterministicmns.py passed, Duration: 39 s
5/179 - wallet_hd.py passed, Duration: 40 s
6/179...
```
After reverting commit:

```$ git revert 04538222
[detached HEAD dc4f4fbe3f] Revert "Merge bitcoin/bitcoin#23799: test: Let test_runner.py start multiple jobs per timeslot"
 1 file changed, 29 insertions(+), 34 deletions(-)
```

Works as expected

```
Early exiting after test failure

TEST                                  | STATUS    | DURATION

rpc_fundrawtransaction.py             | ✓ Passed  | 8 s
wallet_hd.py                          | ✓ Passed  | 8 s
wallet_multiwallet.py --usecli        | ✖ Failed  | 12 s

ALL                                   | ✖ Failed  | 28 s (accumulated) 
Runtime: 12 s
```

Reply:
dropped it